Bug when calling inner function before declaring it

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Linux Mint 20.1
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-3632QM CPU @ 2.20GHz
CC version: cc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0

getwd: /home/fclm/test
vmodules: /home/fclm/.vmodules
vroot: /home/fclm/v
vexe: /home/fclm/v/v
vexe mtime: 2022-09-05 19:23:37
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.1 5e1a2f6.862d91e

Git version: git version 2.25.1
Git vroot status: 0.3.1-24-g862d91ed
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 827f7452

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg test.v`
```v


module main

fn main() {
	
	println(f())

	f := fn () string { return "hello" }
}

```

**What did you expect to see?**

Just "hello" printed.


**What did you see instead?**
```
test.v:9:2: warning: unused variable: `f`
    7 |     println(f())
    8 | 
    9 |     f := fn () string { return "hello" }
      |     ^
   10 | }
==================
/tmp/v_1000/test.12168126026962993754.tmp.c:12073: warning: implicit declaration of function 'f'
/tmp/v_1000/test.12168126026962993754.tmp.c:12073: error: cannot convert 'int' to 'struct string'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -413,6 +413,12 @@ fn (mut c Checker) anon_fn(mut node ast.AnonFn) ast.Type {
 }
 
 pub fn (mut c Checker) call_expr(mut node ast.CallExpr) ast.Type {
+	// Check whether the inner function definition is before the call
+	if var := node.scope.find_var(node.name) {
+		if var.expr is ast.AnonFn && var.pos.pos > node.pos.pos {
+			c.error('unknown function: $node.name', node.pos)
+		}
+	}
 	// TODO merge logic from method_call and fn_call
 	// First check everything that applies to both fns and methods
 	old_inside_fn_arg := c.inside_fn_arg

--- a/vlib/v/checker/tests/inner_functions_call_before_define.out
+++ b/vlib/v/checker/tests/inner_functions_call_before_define.out
@@ -1,0 +1,13 @@
+vlib/v/checker/tests/inner_functions_call_before_define.vv:2:7: error: unknown function: f
+    1 | fn main() {
+    2 |     _ := f()
+      |          ~~~
+    3 |     go f()
+    4 |
+vlib/v/checker/tests/inner_functions_call_before_define.vv:3:5: error: unknown function: f
+    1 | fn main() {
+    2 |     _ := f()
+    3 |     go f()
+      |        ~~~
+    4 | 
+    5 |     f := fn () string { return "hello" }

--- a/vlib/v/checker/tests/inner_functions_call_before_define.vv
+++ b/vlib/v/checker/tests/inner_functions_call_before_define.vv
@@ -1,0 +1,7 @@
+fn main() {
+	_ := f()
+	go f()
+
+	f := fn () string { return "hello" }
+	_ := f() // avoid warnings
+}

